src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolUIItem.cs

```diff
@@ -32,12 +29,10 @@ public bool IsExpanded
         {
             get
             {
-                _threadingContext.ThrowIfNotOnUIThread();
```

Comment:
doc threading.

Reply:
not necessary. We treat these items are pure data, they can be read and written to on any thread.

Reply:
that is exactly what we need to doc.  THis is a mutable type that you are stating can be used on any thread.  That means we can have data races here and we need to doc clearly what the rules are on why things will work out ok here.  Note: i'm not a fan at all of mutable free-threaded types as they are very tricky to get right.  The pattern we use in tagging (which this system is extremely close to) is to build immutable snapshots in the BG, do our diffs in teh BG, then ensure changes to the actual information tied to the UI is handled on a single sync context so mutations don't ever have to worry about races.  :)